packages/ui/src/components/catalog/Catalog.jsx

```diff
@@ -1,53 +1,114 @@
   const [searchTerm, setSearchTerm] = useState("");
   const [isModalOpen, setIsModalOpen] = useState(false);
+  const [totalPages, setTotalPages] = useState(0);
+  const [updateImageId, setUpdateImageId] = useState(null);
 
   const limit = 10;
-  const totalPages = Math.floor(companies.length / 10);
+  const skip = pageNum * limit;
 
-  const skip = pageNum;
-  const skipCount = skip * limit;
+  const { data, makeRequest } = useApi({
+    method: "GET",
+    url: `/catalog/logos?skip=${skip}&limit=${limit}`,
+  });
 
-  const companiesInfo = companies.slice(
-    skipCount,
-    Math.min(skipCount + limit, companies.length)
-  );
-  const filteredCompanies = companiesInfo.filter((company) =>
-    company.companyImage.toLowerCase().includes(searchTerm.toLowerCase())
-  );
+  const { loading: uploadLoading, makeRequest: uploadMakeRequest } = useApi({
+    method: updateImageId ? "PUT" : "POST",
+    url: `/catalog/logo`,
+    headers: { "Content-Type": "multipart/form-data" },
+  });
+
+  useEffect(() => {
+    makeRequest();
+  }, [pageNum]);
+
+  useEffect(() => {
+    if (data?.data?.totalPages) {
+      setTotalPages(data.data.totalPages - 1);
+    }
+  }, [data]);
+
+  const handleImageUpload = async ({ file, companyUri }) => {
+    if (!file || !companyUri) return;
+
+    const formData = new FormData();
+    formData.append("logo", file);
+    formData.append("companyUri", companyUri);
+
+    try {
+      const success = await uploadMakeRequest({ data: formData });
+      if (success) {
+        setIsModalOpen(false);
+        makeRequest();
+      }
+    } catch (err) {
+      console.error("Upload error:", err);
+    }
+  };
+
+  const handleUpdateImage = async ({ file }) => {
+    if (!file || !updateImageId) return;
+
+    const formData = new FormData();
+    formData.append("logo", file);
+    formData.append("id", updateImageId);
+
+    try {
+      const success = await uploadMakeRequest({ data: formData });
+      if (success) {
+        setIsModalOpen(false);
+        makeRequest();
+      }
+    } catch (err) {
+      console.error("Upload error:", err);
+    } finally {
+      setUpdateImageId(null);
+    }
+  };
 
   const handleSearchTermChange = (inputChangeEvent) => {
     setSearchTerm(inputChangeEvent.target.value);
   };
 
   const handlePreviousBtnClick = () => {
-    if (pageNum == 0) {
-      return;
+    if (pageNum > 0) {
+      setSearchTerm("");
+      setPageNum((prev) => prev - 1);
     }
-    setSearchTerm("");
-    setPageNum((prevPageNum) => prevPageNum - 1);
   };
 
   const handleNextBtnClick = () => {
-    if (pageNum == totalPages) {
-      return;
+    if (pageNum < totalPages) {
+      setSearchTerm("");
+      setPageNum((prev) => prev + 1);
     }
-    setSearchTerm("");
-    setPageNum((prevPageNum) => prevPageNum + 1);
   };
 
+  const handleReuploadBtnClick = (id) => {
+    setIsModalOpen(true);
+    setUpdateImageId(id);
+  };
+
+  const filteredCompanies =
```

Comment:
better to have this function wrapped inside `useMemo` so that we don't recompute it every time